pkg/serdes.go

```diff
@@ -37,83 +37,112 @@ func readBodyBuffered(resp *http.Response) ([]byte, error) {
 }
 
 func imageFromBuffer(buffer []byte, transform *ARAwareTransform, aspectRatio float64, encodeImage bool, isMask bool) (*ImagePayload, float64, error) {
-	// Decode the image payload using vips
-	img, err := vips.NewImageFromBuffer(buffer)
+	// Decode the image payload using vips, using bulletproof settings
+	importParams := vips.NewImportParams()
+	importParams.AutoRotate.Set(true)
+	importParams.FailOnError.Set(true)
+	importParams.Page.Set(0)
+	importParams.NumPages.Set(1)
+	importParams.HeifThumbnail.Set(false)
+	importParams.SvgUnlimited.Set(false)
+
+	img, err := vips.LoadImageFromBuffer(buffer, importParams)
 	if err != nil {
-		return nil, -1., err
+		return nil, -1., fmt.Errorf("error loading image: %w", err)
 	}
 
-	err = img.AutoRotate()
-	if err != nil {
-		return nil, -1., err
-	}
-
-	// Optionally crop and resize the image on the fly. Save the aspect ratio in the process for future use
-	originalWidth, originalHeight := img.Width(), img.Height()
-
-	if transform != nil {
-		aspectRatio, err = transform.cropAndResizeToClosestAspectRatio(img, aspectRatio)
-		if err != nil {
-			return nil, -1., err
-		}
-	}
-
-	width, height := img.Width(), img.Height()
-
 	// If the image is 4 channels, we need to drop the alpha channel
 	if img.Bands() == 4 {
 		err = img.Flatten(&vips.Color{R: 255, G: 255, B: 255}) // Flatten with white background
 		if err != nil {
-			fmt.Println("Error flattening image:", err)
-			return nil, -1., err
+			return nil, -1., fmt.Errorf("error flattening image: %w", err)
 		}
 		fmt.Println("Image flattened")
 	}
 
 	// If the image is not a mask but is 1 channel, we want to convert it to 3 channels
 	if (img.Bands() == 1) && !isMask {
-		err = img.ToColorSpace(vips.InterpretationSRGB)
+		if img.Metadata().Format == vips.ImageTypeJPEG {
+			err = img.ToColorSpace(vips.InterpretationSRGB)
+			if err != nil {
+				return nil, -1., fmt.Errorf("error converting to sRGB: %w", err)
+			}
+		} else {
+			// // FIXME: maybe that we could recover these still. By default throws an error, sRGB and PNG not supported
```

Comment:
cleaner conversion, vips would complain that we were doing something illegal for these images so skipping it now